### PR TITLE
layers: Add VK_KHR_dynamic_rendering VUs - Change Set #2

### PR DESCRIPTION
**Pull request by AaronHaganAMD**
Add additional VUs for the VK_KHR_dynamic_rendering extension.
Small refactors and unit tests as well as addition of  VUIDs for
VkDraw* commands.

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
Looks like this is failing on our NV test machines:
```
[ RUN      ] VkPositiveLayerTest.DynamicRenderingDraw
ERROR: -------------------------------------------------------------
ERROR: FAIL during test after 18.22 minutes (1092.95 seconds) with execute(C:\j\w\64-bit\Debug\Vulkan-ValidationLayers\dbuild\tests\Debug\vk_layer_validation_tests.exe --gtest_catch_exceptions=0 --gtest_filter=-VkLayerTest.NonSemanticInfoEnabled) failed after 18.22 minutes (1092.92 seconds) with returncode=3221225477 (0xc0000005=STATUS_ACCESS_VIOLATION)
INFO: Test run complete after 23.60 minutes (1415.82 seconds)
```

File: tests/positive/dynamic_rendering.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+// POSITIVE VALIDATION TESTS
+//
+// These tests do not expect to encounter ANY validation errors pass only if this is true
+
+TEST_F(VkPositiveLayerTest, DynamicRenderingDraw) {
+    TEST_DESCRIPTION("Draw with Dynamic Rendering.");
+
+    SetTargetApiVersion(VK_API_VERSION_1_1);
+    ASSERT_NO_FATAL_FAILURE(InitFramework());
+    if (DeviceExtensionSupported(VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME)) {
```
Comment (jeremyg-lunarg) [inline tests/positive/dynamic_rendering.cpp:49]: This should use AddRequiredDeviceExtensions() I need to fix my test case to use it too

**Review by jeremyg-lunarg — CHANGES_REQUESTED** [review]
Can you please put the test code into a separate git commit? 

The validation code looks reasonable, I think.

I can try to reproduce the NV crash later today  and get more info on what is wrong there.


Thanks!

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -654,6 +654,60 @@ bool CoreChecks::ValidatePipelineDrawtimeState(const LAST_BOUND_STATE &state, co
     const DrawDispatchVuid vuid = GetDrawDispatchVuid(cmd_type);
     const char *caller = CommandTypeString(cmd_type);
 
+    if (pPipeline->rp_state->use_dynamic_rendering) {
+        if (pPipeline->rp_state->dynamic_rendering_pipeline_create_info.viewMask !=
+            pCB->activeRenderPass->dynamic_rendering_begin_rendering_info.viewMask) {
+                skip |= LogError(pCB->commandBuffer(), vuid.dynamic_rendering_view_mask,
+                    "%s: Currently bound pipeline %s viewMask (%u) must be equal to pBeginRendering->viewMask (%u)",
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:661]: Can `%u` be replaced with `PRIu32` (or the appropriate type)? We have standardized on using the `PRI*` macros to give us some "type safety" with printf.

**Review by ncesario-lunarg — COMMENTED** [review]
One nit with the print formatting; otherwise I echo @jeremyg-lunarg's comments.

**Review by jeremyg-lunarg — COMMENTED** [review]
LGTM but we still need to update a driver in CI before the new test can merge

**Comment by jeremyg-lunarg** [comment]
CI failure today is a different one. Looks like many of the log messages in CoreChecks::ValidatePipelineDrawtimeState() are now using strings but have % PRIu32 formats rather than %s:
```
layers/core_validation.cpp: In member function ‘bool CoreChecks::ValidatePipelineDrawtimeState(const LAST_BOUND_STATE&, const CMD_BUFFER_STATE*, CMD_TYPE, const PIPELINE_STATE*) const’:
layers/core_validation.cpp:722:33: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 7 has type ‘const char*’ [-Wformat=]
  722 |                                 "%s: Color attachment (%" PRIu32 ") sample count (%u) must match corresponding VkAttachmentSampleCountInfoAMD sample count (%" PRIu32 ")",
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  723 |                                 caller, i, string_VkSampleCountFlagBits(color_image_samples),
      |                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                        |
      |                                                                        const char*
layers/core_validation.cpp:722:33: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 8 has type ‘const char*’ [-Wformat=]
  722 |                                 "%s: Color attachment (%" PRIu32 ") sample count (%u) must match corresponding VkAttachmentSampleCountInfoAMD sample count (%" PRIu32 ")",
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  723 |                                 caller, i, string_VkSampleCountFlagBits(color_image_samples),
  724 |                                 string_VkSampleCountFlagBits(p_attachment_sample_count_info_amd->pColorAttachmentSamples[i]));
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             |

```

**Comment by AaronHaganAMD** [comment]
> CI failure today is a different one. Looks like many of the log messages in CoreChecks::ValidatePipelineDrawtimeState() are now using strings but have % PRIu32 formats rather than %s:
> 
> ```
> layers/core_validation.cpp: In member function ‘bool CoreChecks::ValidatePipelineDrawtimeState(const LAST_BOUND_STATE&, const CMD_BUFFER_STATE*, CMD_TYPE, const PIPELINE_STATE*) const’:
> layers/core_validation.cpp:722:33: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 7 has type ‘const char*’ [-Wformat=]
>   722 |                                 "%s: Color attachment (%" PRIu32 ") sample count (%u) must match corresponding VkAttachmentSampleCountInfoAMD sample count (%" PRIu32 ")",
>       |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>   723 |                                 caller, i, string_VkSampleCountFlagBits(color_image_samples),
>       |                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>       |                                                                        |
>       |                                                                        const char*
> layers/core_validation.cpp:722:33: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 8 has type ‘const char*’ [-Wformat=]
>   722 |                                 "%s: Color attachment (%" PRIu32 ") sample count (%u) must match corresponding VkAttachmentSampleCountInfoAMD sample count (%" PRIu32 ")",
>       |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>   723 |                                 caller, i, string_VkSampleCountFlagBits(color_image_samples),
>   724 |                                 string_VkSampleCountFlagBits(p_attachment_sample_count_info_amd->pColorAttachmentSamples[i]));
>       |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>       |                                                             |
> ```

these should be fixed now.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM. Linux github failures are due to a new Vulkan-Loader change which is causing the same failure in other PRs.

**Pull request merged**